geode-core/src/test/java/org/apache/geode/internal/cache/PRHARedundancyProviderTest.java

```diff
@@ -14,30 +14,36 @@
  */
 package org.apache.geode.internal.cache;
 
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
 import static org.mockito.Mockito.doReturn;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
+import java.util.TreeSet;
+
 import org.junit.Before;
 import org.junit.Test;
 
+import org.apache.geode.internal.cache.partitioned.InternalPRInfo;
+import org.apache.geode.internal.cache.partitioned.LoadProbe;
 import org.apache.geode.internal.cache.partitioned.PersistentBucketRecoverer;
 
 
 public class PRHARedundancyProviderTest {
   private PRHARedundancyProvider provider;
+  private PartitionedRegion mockPartitionedRegion;
 
   @Before
   public void setup() {
-    PartitionedRegion partitionedRegion = mock(PartitionedRegion.class, RETURNS_DEEP_STUBS);
+    mockPartitionedRegion = mock(PartitionedRegion.class, RETURNS_DEEP_STUBS);
     InternalCache cache = mock(InternalCache.class);
     DistributedRegion root = mock(DistributedRegion.class);
-    when(partitionedRegion.getCache()).thenReturn(cache);
+    when(mockPartitionedRegion.getCache()).thenReturn(cache);
     when(cache.getRegion(PartitionedRegionHelper.PR_ROOT_REGION_NAME, true)).thenReturn(root);
-    provider = spy(new PRHARedundancyProvider(partitionedRegion));
+    provider = spy(new PRHARedundancyProvider(mockPartitionedRegion));
```

Comment:
Any reason for making this a spy? I didn't see one.
If not, then I think it is better for "provider" to be not be a spy since it is the class we are unit testing.
When I see a spy I then want to know what part of the class we are testing is being mocked.

Reply:
the other tests (not part of this commit) were mocking `getPersistentBucketRecoverer`

not sure i want to unwind that in this commit.